patches/minecraft/net/minecraft/world/entity/item/ItemEntity.java.patch

```diff
@@ -144,3 +145,88 @@
+    public float m_213816_() {
+       return 180.0F - this.m_32008_(0.5F) / ((float)Math.PI * 2F) * 360.0F;
+    }
++
++   // FORGE START
++   private final java.util.Set<net.minecraftforge.items.pickup.ItemPickupReason> disabledPickupReasons = com.google.common.collect.Sets.newHashSet();
++   private final java.util.Set<net.minecraftforge.items.pickup.ItemPickupQuery> pickupQueries = com.google.common.collect.Sets.newHashSet();
++
++   /**
++    * Disallow item from being picked up for the given {@link net.minecraftforge.items.pickup.ItemPickupReason reason}
++    *
++    * @param pickupReason The reason to disallow pick-ups for
++    */
++   public final void disablePickupReason(net.minecraftforge.items.pickup.ItemPickupReason pickupReason)
++   {
++      disabledPickupReasons.add(pickupReason);
++   }
++
++   /**
++    * Allow item to be picked up for the given {@link net.minecraftforge.items.pickup.ItemPickupReason reason}
++    *
++    * @param pickupReason The reason to allow pick-ups for
++    */
++   public final void enablePickupReason(net.minecraftforge.items.pickup.ItemPickupReason pickupReason)
++   {
++      disabledPickupReasons.remove(pickupReason);
++   }
++
++   /**
++    * Add a {@link net.minecraftforge.items.pickup.ItemPickupQuery query} to validated when attempting to pick up this item.
++    *
++    * @param pickupQuery Query to be validated when this item is being picked up
++    */
++   public final void addPickupQuery(net.minecraftforge.items.pickup.ItemPickupQuery pickupQuery)
++   {
++      pickupQueries.add(pickupQuery);
++   }
++
++   /**
++    * Remove the {@link net.minecraftforge.items.pickup.ItemPickupQuery query} from being validated when this item is being picked up
++    *
++    * @param pickupQuery Query to no longer be validated when this item is being picked up
++    */
++   public final void removePickupQuery(net.minecraftforge.items.pickup.ItemPickupQuery pickupQuery)
++   {
++      pickupQueries.remove(pickupQuery);
++   }
++
++   /**
++    * Returns true if this item is allowed to be picked up for the given {@link net.minecraftforge.items.pickup.ItemPickupReason reasons}
++    *
++    * @param collector The object attempting to pick up this item
++    * @param pickupReasons The reasons this item is being picked up
++    * @return if this item is allowed to be picked up for the given {@link net.minecraftforge.items.pickup.ItemPickupReason reasons}
++    */
++   public final boolean isPickupAllowed(Object collector, net.minecraftforge.items.pickup.ItemPickupReason... pickupReasons)
++   {
++      // check forge event to see if item is forcibly allowed to be picked up or not
++      var event = new net.minecraftforge.event.entity.item.ItemPickupAllowedEvent(this, collector, pickupReasons);
++      net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(event);
++      var eventResult = event.getResult();
++
++      if(eventResult == net.minecraftforge.eventbus.api.Event.Result.ALLOW) return true;
```

Comment:
Formatting: space between `if` keyword and opening parenthesis. (Same comment for the rest of this method.)